skimage/exposure/_adapthist.py

```diff
@@ -326,36 +308,32 @@ def interpolate(image, slices, maps, lut):
 
     Notes
     -----
-    This function calculates the new graylevel assignments of pixels within
-    a submatrix of the image. This is done by bilinear interpolation between
-    2^image.ndim different adjacent mappings in order to eliminate boundary
-    artifacts.
+    This function calculates the new graylevel assignments of pixels
+    within a submatrix of the image. This is done by multilinear
+    interpolation between 2^image.ndim different adjacent mappings
+    in order to eliminate boundary artifacts.
     """
-    norm = np.product([slices[dim].size
-                       for dim in range(image.ndim)])  # Normalization factor
+    region = tuple([s for s in slices])
+    view = image[region]
 
     # interpolation weight matrices
-    coeffs = np.meshgrid(*tuple([np.arange(slices[dim].size)
-                                 for dim in range(image.ndim)]), indexing='ij')
-    coeffs = [coeff.transpose() for coeff in coeffs]
+    coeffs = np.meshgrid(*tuple([np.arange(s) for s in view.shape[::-1]]),
+                         indexing='ij')
+    coeffs = [np.transpose(c) for c in coeffs]
 
-    inv_coeffs = [np.flip(coeffs[dim], axis=image.ndim - dim - 1) + 1
-                  for dim in range(image.ndim)]
-
-    region = tuple([slice(int(slices[dim][0]), int(slices[dim][-1] + 1))
-                    for dim in range(image.ndim)][::-1])
-    view = image[region]
+    inv_coeffs = [np.flip(c, axis=image.ndim - dim - 1) + 1
+                  for dim, c in enumerate(coeffs)]
 
     im_slice = lut[view]
 
-    new = np.zeros_like(view, dtype=int)
+    result = np.zeros_like(view, dtype=int)
     for iedge, edge in enumerate(np.ndindex(*([2] * image.ndim))):
-        edge = edge[::-1]
-        new += \
-            np.product([[inv_coeffs, coeffs][edge[dim]][dim]
-                        for dim in range(image.ndim)], 0)\
-            * maps[iedge][im_slice]
-
-    new = (new / norm).astype(view.dtype)
-    view[::] = new
+        result += (np.product([[inv_coeffs, coeffs][e][dim]
+                               for dim, e in enumerate(edge[::-1])], 0)
+                   * maps[iedge][im_slice])
+
+    # normalize
+    result = result / np.product(view.shape)
```

Comment:
```suggestion
    result /= view.size
```

Reply:
Wasn't aware of this attribute! Let's assume I'm the only one :P